tools/public_api_guard/material/core.d.ts

```diff
@@ -87,6 +89,8 @@ export declare class DateRange<D> {
     end: D | null);
 }
 
+export declare type DateRangeSelectionStrategy<D> = (selectedDate: D | null, currentRange: DateRange<D>, adapter: DateAdapter<D>) => DateRange<D>;
```

Comment:
Should this follow the same global options pattern we've used in other components, i.e. make it `DateRangePickerOptions`

Reply:
I suppose, but I don't see what else we might attach to that object.

Reply:
For now nothing, but that's the pattern we've been using because its easy to expand the API later on should the need arise